pkg/apis/beat/v1beta1/beat_types.go

```diff
@@ -79,7 +79,8 @@ type BeatSpec struct {
 }
 
 type DaemonSetSpec struct {
-	PodTemplate corev1.PodTemplateSpec `json:"podTemplate,omitempty"`
+	PodTemplate corev1.PodTemplateSpec         `json:"podTemplate,omitempty"`
+	Strategy    appsv1.DaemonSetUpdateStrategy `json:"strategy,omitempty"`
```

Comment:
I guess it's not required, but maybe add `// +kubebuilder:validation:Optional` for consistency with other fields.